src/server.c

```diff
@@ -4829,7 +4926,7 @@ sds genRedisInfoString(const char *section) {
     }
 
     /* Memory */
-    if (allsections || defsections || !strcasecmp(section,"memory")) {
+    if (!server.sentinel_mode && (all_sections || (dictFind(section_dict,"memory") != NULL))) {
```

Comment:
if you follow my advise, and remove have sentinel sanitize the section list before calling this function, you will not need the sentinel check here, and in any other sections in this function.
(of course it should avoid passing the all or everything flag, and instead just set a static list of sections in the dict)
```suggestion
    if (all_sections || (dictFind(section_dict,"memory") != NULL)) {
```